core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -388,7 +388,7 @@ class Partition(val topic: String,
     val leaderHWIncremented = inWriteLock(leaderIsrUpdateLock) {
       leaderReplicaIfLocal match {
         case Some(leaderReplica) =>
-          val outOfSyncReplicas = getOutOfSyncReplicas(leaderReplica, replicaMaxLagTimeMs)
+          val outOfSyncReplicas = getOutOfSyncReplicas(leaderReplica, replicaMaxLagTimeMs).filter(r => !r.ignoreIsrCheck)
```

Comment:
@junrao Thanks for the suggestion.
I gave it a go and it makes our test pass without the need for ignoreIsrCheck() but it breaks a number of tests in UncleanLeaderElectionTest and ReplicationQuotasTest. I'm not too familiar with these tests and couldn't tell immediately what's going on. I'll have a closer look see if I can understand what breaks them